### ISSUE TITLE
Pool workers live on after process exit

### ISSUE THREAD

**Issue by hirikarate** [andywer/threads.js#147]
If I don't call
```
await Thread.terminate(add)
```
or
```
await pool.terminate()
```
when my application exits forcefully or crashes, will the running worker threads and those in pool still exist and cause resource leaking? I can't find this information in documentation.

Thank you.

**Comment by andywer** [comment]
Hey @hirikarate!

No, actually it's supposed to kill the remaining workers when the process exits.

Can you please tell me what node version you are using and how you quit the process when that happens? You mentioned forceful quitting of the process. I suppose this only happens when the process does not exit normally.

**Comment by hirikarate** [comment]
This is actually a starter's question since I am new to thread.js. This is not a bug.

I just asked to know if I need to handle my class' destruction, and listen to signal SIGTERM and SIGINT.

```
const pool = this._pool = Pool(() => spawn(Worker('./rsa-jwt-worker')), 8)

class JwtHelper {
  /*
   * Necessary or not?
   */
  static destroy() {
    return pool.terminate()
  }

  encodeToken(options) {
    return pool.queue((jwtFn) => jwtFn.sign(options))
  }

  decodeToken(options) {
    return pool.queue((jwtFn) => jwtFn.verify(options))
  }
}

**Comment by andywer** [comment]
It kind of is a bug, since we never want to leave orphaned child processes behind :)

You only need to handle your class' destruction if you want to be able to kill the pool and still keep running the app. Normally, all worker processes should be killed upon process exit.

Can you please state the node version you are using, though? (`node --version`)

**Comment by hirikarate** [comment]
My Node version is v12.8.0, development on Windows 10, and plan to go live in CentOS image of Docker container.

I have not noticed any live-on / leaked threads, this question is just out of my curiosity because I don't find it in documentation.

Thank you for your quick answers, I will inform if I find something.

**Comment by andywer** [comment]
I prepared a little fix that will hopefully solve the issue.

Can you install `threads@1.0.0-beta.3-bugfix-147` and try again?

**Comment by andywer** [comment]
Published the fix as `1.0.0-beta.4`. Will hopefully fix the issue - feel free to re-open the issue if not 😉
